EventReader/src/Event/EventStream.cs

```diff
@@ -0,0 +1,57 @@
+﻿using EventReader.Read;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace EventReader.Event
+{
+    public class EventStream
+    {
+        IDataSource Source;
+
+        public EventStream(IDataSource source)
+        {
+            Source = source;
+        }
+
+        public IEnumerable<IEvent> Get(string query)
+        {
+            foreach(EventEntry entry in Reader.Read(Source, query)) // TODO: work out how to segment collection
```

Comment:
this foreach loop is just projecting each element from the Reader's enumerable, there's a functional operator to do that - `Select`